tests/samplers_tests/tpe_tests/test_multi_objective_sampler.py

```diff
@@ -489,28 +489,45 @@ def test_multi_objective_sample_independent_ignored_states() -> None:
     assert len(set(suggestions)) == 1
 
 
-def test_multi_objective_get_observation_pairs() -> None:
+@pytest.mark.parametrize("constraints_enabled", [False, True])
+@pytest.mark.parametrize("constraint_value", [-2, 2])
+def test_multi_objective_get_observation_pairs(
```

Comment:
I think we can just add another test function to improve the readability of unit tests. It slightly increase the code redundancy, but it can be accepted. What do you think?

Reply:
Umm, this is also not a strong opinion, but I think if there are multiple tests on the same function, the fixing cost is high when the interface of the original function changes. Which part of the current implementation do you think is hindering readability?
Also, do you have the same opinion on other unit tests on `test_sampler.py`?

Reply:
> Which part of the current implementation do you think is hindering readability?

My personal opinion is that constraint setup (e.g., [this one](https://github.com/optuna/optuna/pull/3506/files#diff-e52a24e2c14ee4ff0d8b5e62f38c4e515ec3dad12446570f0771ef74d27689faR502)) is required even for tests that have nothing to do with constrained optimization.

I think a test should be in a form that is immediately understandable in itself and should not have complex logic inside. For example, but what about cutting out the parts that can be commonly used by constrained optimization tests and existing tests as utilities and referencing them from separate test functions? (I know I said so myself, but this is going to be a major undertaking, so maybe we should tackle this in a separate PR.)

Reply:
I understand your point.

> what about cutting out the parts that can be commonly used by constrained optimization tests and existing tests as utilities and referencing them from separate test functions?

Then, could I work on the above suggestion after this PR is merged?

Reply:
I agree with @HideakiImamura . This test changes the expected values depending on the parametrized value (i.e., `constraints_enabled`), and I think it is a kind of [conditional test logic](http://xunitpatterns.com/Conditional%20Test%20Logic.html). As you mentioned, we can work on it in a follow-up PR.